projects/canopy/src/lib/forms/input/input-field.component.ts

```diff
@@ -47,28 +80,108 @@ export class LgInputFieldComponent {
     this._inputElement = element;
     this._inputElement.id = this.id;
   }
+  get inputElement() {
+    return this._inputElement;
+  }
 
-  _hintElement: LgHintComponent;
+  private _hintElement: LgHintComponent;
   @ContentChild(LgHintComponent)
   set hintElement(element: LgHintComponent) {
-    this._inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
-      this._inputElement.ariaDescribedBy,
+    this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.inputElement.ariaDescribedBy,
       this._hintElement,
       element,
     );
     this._hintElement = element;
   }
 
-  _validationElement: LgValidationComponent;
+  private _validationElement: LgValidationComponent;
   @ContentChild(LgValidationComponent)
   set errorElement(element: LgValidationComponent) {
-    this._inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
-      this._inputElement.ariaDescribedBy,
+    this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.inputElement.ariaDescribedBy,
       this._validationElement,
       element,
     );
     this._validationElement = element;
   }
 
+  @ContentChild(LgButtonComponent) buttonElement: LgButtonComponent;
+
+  private _suffixChildren: QueryList<LgSuffixDirective>;
+  @ContentChildren(LgSuffixDirective)
+  set suffixChildren(elements: QueryList<LgSuffixDirective>) {
+    elements.forEach((element) => {
+      this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+        this.inputElement.ariaDescribedBy,
+        this._validationElement,
+        element,
+      );
+    });
+    this._suffixChildren = elements;
+  }
+  get suffixChildren() {
+    return this._suffixChildren;
+  }
+
+  private _prefixChildren: QueryList<LgSuffixDirective>;
+  @ContentChildren(LgPrefixDirective)
+  set prefixChildren(elements: QueryList<LgSuffixDirective>) {
+    elements.forEach((element) => {
+      this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+        this.inputElement.ariaDescribedBy,
+        this._validationElement,
+        element,
+      );
+    });
+    this._prefixChildren = elements;
+  }
+  get prefixChildren() {
+    return this._prefixChildren;
+  }
+
+  @Input() id = `lg-input-${nextUniqueId++}`;
```

Comment:
I think it would be better to save `nextUniqueId++` into a separate const and then use it, otherwise if we end up reusing it, it will increase each time.

Reply:
This was mostly reformatting and not changed in the PR. I'm open to changing it though but I'm not sure I quite follow. The idea is that it should increase each time, it was a convention I shamefully [borrowed from material ui](https://github.com/angular/components/blob/master/src/material/input/input.ts#L94)

Reply:
Yes, it should increase each time a new component is instantiated.
I have had to fix multiple instances where we did the following mistake: 
```
@Input() id = `lg-input-${nextUniqueId++}`;
anotherVariable = `my-variable-${nextUniqueId++}`
```
which produces:
```
@Input() id = `lg-input-1`;
anotherVariable = `my-variable-2`
```
while it should be:
```
@Input() id = `lg-input-1`;
anotherVariable = `my-variable-1`
```

I know in this component this is not an issue atm but it did happen multiple times and I'm just concerned that someone might come along and do the same.

Reply:
ah ok that makes sense, so in theory with the first example it still works as the IDs are unique, but it's harder to ascertain which elements join together as the numerical part of the id is different. (pushed a change)